lib/auth/auth.js

```diff
@@ -111,11 +113,14 @@ function doAuth(request, log, cb, awsService, requestContexts) {
         return cb(null, res.params);
     }
 
-    requestContexts.forEach(requestContext => {
-        requestContext.setAuthType(res.params.data.authType);
-        requestContext.setSignatureVersion(res.params.data.signatureVersion);
-        requestContext.setSignatureAge(res.params.data.signatureAge);
-    });
+    if (requestContexts) {
```

Comment:
So I take it that the batch delete only tries to authenticate, and not check that policies allow the user to delete ?
If not, could you clarify the cases in some sort of doc/comment ?